File: src/main/java/poetry/database/DatabaseConfiguration.java
package poetry.database;

public class DatabaseConfiguration {
	public static final String DEFAULT_NAME = "database";
	private final int modelVersion;
	private final Class<?>[] modelClasses;
	private final String databaseName;

	public DatabaseConfiguration(int modelVersion, Class<?>[] modelClasses, String databaseName) {
		this.modelVersion = modelVersion;
		this.modelClasses = modelClasses;
		this.databaseName = databaseName;
	}

	public DatabaseConfiguration(int modelVersion, Class<?>[] modelClasses) {
		this(modelVersion, modelClasses, DEFAULT_NAME);
	}


	public int getModelVersion() {
		return modelVersion;
	}

	public Class<?>[] getModelClasses() {
		return modelClasses;
	}

	public String getDatabaseName() {
		return databaseName;
	}
}

File: src/main/java/poetry/DatabaseConfiguration.kt
package poetry

import kotlin.reflect.KClass

class DatabaseConfiguration @JvmOverloads constructor(
	val modelVersion: Int,
	val modelClasses: Array<Class<*>>,
	val databaseName: String = DEFAULT_NAME
) {
	companion object {
		const val DEFAULT_NAME = "database"
	}
}

fun databaseConfigurationOf(dbName: String, vararg types: KClass<*>): DatabaseConfiguration {
	val classList = types.map { it.java }
	return DatabaseConfiguration(1, classList.toTypedArray(), dbName)
}

fun databaseConfigurationOf(dbName: String, vararg types: Class<*>): DatabaseConfiguration {
	return DatabaseConfiguration(1, arrayOf(*types), dbName)
}
